src/components/themed/WalletSyncCircle.js

```diff
@@ -18,6 +18,15 @@ type Props = {
   wallet: EdgeCurrencyWallet
 }
 
+// We are "done" once the sync ratio exceeds this value.
+// We want to tolerate floating-point rounding errors,
+// so this is a bit less than 1:
+const DONE_VALUE = 0.999
```

Comment:
Is this really needed? I'm assuming we'll run into floating point issues when we exceed `Number.MAX_SAFE_INTEGER`, but how likely are wallet plugins going to be dealing with such large numbers in the ratio calculation. And even if they were to incorrectly report `<1` at full-sync, this prompts a fix within the plugin.

The tolerance introduces a bug where a UTXO-based plugin reports `0.9997` sync ratio because the last address has many transactions to process. This will rarely occur for many wallets, but for large wallets this could be a real issue (recently came into contact with a wallet which had the last address with more than 4000 transactions).

I may be making a mountain out of a mole-hill?

Reply:
Try plugging `(0.1+0.1+0.1)/0.3` into node.js. You will get `1.0000000000000002` as an answer, so these effects are real and pretty easy to get. I could see using `0.999999999999` as the threshold - it's got 12 digits of precision, which is a bit less than floating point's ~15 digits.

The old redux-based version also had this logic, but there the threshold was 0.95. This is why we have never seen the problem in the wild before - our app always had mitigations in place.

Reply:
I'll secede from my position for the sake of pushing through with this PR.